app/views/protocols/import_export/_import_json_protocol_preview_modal.html.erb

```diff
@@ -0,0 +1,338 @@
+
+                  <% else %>
+
+                    <% end #case if%>
+
+                    <% end #inner if%>
+
+                    <% end #step component loop %>
+
+                  </div>
+                  <hr>
+                  <!-- This could be implemented later maybe,i believe this is different styling
+                  for scinotes own rich text format tables
+
+                     <div class="row">
+              <div data-toggle="tables" class="col-xs-12" style="display: none;">
+                <div data-hold="tables"> </div>
+              </div>
+              <div data-toggle="assets" class="col-xs-12" style="display: none;">
+                <strong>Files</strong>
+                <ul data-hold="assets">
+                </ul>
+              </div>
+              <div class="col-xs-12">
+                <div data-hold="checklists"></div>
+              </div>
+            </div> -->
+                </div>
+              </div>
+
+            </div>
+
+            <% end  #step loop%>
+
+            <% if
+            @json_object["manuscript_citation"]&&@json_object["manuscript_citation"]!="" %>
+
+            <br>
+            <strong>Manuscript citation:</strong><br>
+            <%= (@json_object["manuscript_citation"]) %>
+
+            <% end %>
+            <% if @json_object["publish_date"]&&@json_object["publish_date"]!="" %>
+
+            <br>
+            <strong>Publish date:</strong><br>
+            <%= (@json_object["publish_date"]) %>
+
+            <% end %>
+            <% if @json_object["vendor_name"]&&@json_object["vendor_name"]!="" %>
+
+            <br>
+            <strong>Vendor name:</strong><br>
+            <%= (@json_object["vendor_name"]) %>
+
+            <% end %>
+            <% if @json_object["vendor_link"]&&@json_object["vendor_link"]!="" %>
+
+            <br>
+            <strong>Vendor link:</strong><br>
+            <%= (@json_object["vendor_link"]) %>
+
+            <% end %>
+            <% if @json_object["keywords"]&&@json_object["keywords"]!="" %>
+
+            <br>
+            <strong>Keywords:</strong><br>
+            <%= (@json_object["keywords"]) %>
+
+            <% end %>
+            <% if @json_object["tags"]&&@json_object["tags"]!="" %>
+
+            <br>
+            <strong>Tags:</strong><br>
+            <%  @json_object["tags"].each do |tag| %>
+            <%= (tag["tag_name"])+" , " %>
+            <% end %>
+
+            <% end %>
+            <hr>
+          </div>
+
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-default" data-dismiss="modal"><%= t("general.cancel") %></button>
+
+          <%= f.submit t("protocols.import_export.import_modal.import"),class: "btn
+          btn-primary" %>
+          <% end %>
+
+        </div>
+      </div>
+    </div>
+  </div>
+  <%# the below script resets all the fields after you decide not to import the
+  file (cancel (modal hide)) %>
```

Comment:
You should put these script in the js file.

Reply:
Since the page that this modal is rendered on is index, i should put them in index.js correct? (app/assets/javascripts/protocols/index.js)